pkg/core/blockchain.go

```diff
@@ -699,160 +738,180 @@ func (bc *Blockchain) storeBlock(block *block.Block, txpool *mempool.Pool) error
 	}
 	writeBuf.Reset()
 
-	aer, err := bc.runPersist(bc.contracts.GetPersistScript(), block, cache, trigger.OnPersist)
-	if err != nil {
-		return fmt.Errorf("onPersist failed: %w", err)
-	}
-	appExecResults = append(appExecResults, aer)
-	err = cache.PutAppExecResult(aer, writeBuf)
-	if err != nil {
-		return fmt.Errorf("failed to store onPersist exec result: %w", err)
+	canStartSync := bc.canStartSync()
+	if canStartSync {
```

Comment:
Maybe we can add some `PreBlockchain` to not break `Blockchain` proper and only create full `Blockchain` when we have proper ledger state?

Reply:
I added a separate `saveBlock` method not to break `storeBlock`, which seems to be prettier than before.